api/securitytest/spotbugs.go

```diff
@@ -0,0 +1,204 @@
+	if errorGradleFound {
+		spotbugsScan.ErrorFound = errors.New("error running gradle")
+		spotbugsScan.prepareSpotBugsVulns()
+		spotbugsScan.prepareContainerAfterScan()
+		return nil
+	}
+
+	// check if Maven failed running
+	errorMavenFound := strings.Contains(spotbugsScan.Container.COutput, "ERROR_RUNNING_MAVEN_BUILD")
+	if errorMavenFound {
+		spotbugsScan.ErrorFound = errors.New("error running maven")
+		spotbugsScan.prepareSpotBugsVulns()
+		spotbugsScan.prepareContainerAfterScan()
+		return nil
+	}
+
+	// check if unsuported java project was found
+	errorUnsuported := strings.Contains(spotbugsScan.Container.COutput, "ERROR_UNSUPPORTED_JAVA_PROJECT")
+	if errorUnsuported {
+		spotbugsScan.ErrorFound = errors.New("error unsuported java project")
+		spotbugsScan.prepareSpotBugsVulns()
+		spotbugsScan.prepareContainerAfterScan()
+		return nil
+	}
+
+	// nil cOutput states that no Issues were found.
+	if spotbugsScan.Container.COutput == "" {
+		spotbugsScan.prepareContainerAfterScan()
+		return nil
+	}
+
+	// Unmarshall rawOutput into finalOutput, that is a SpotBugsOutput struct.
+	spotBugsOutput, err := parseXMLtoJSON([]byte(spotbugsScan.Container.COutput))
+	if err != nil {
+		log.Error("analyzeSpotBugs", "SPOTBUGS", 1039, spotbugsScan.Container.COutput, err)
+		spotbugsScan.ErrorFound = err
+		spotbugsScan.prepareContainerAfterScan()
+		return err
+	}
+
+	spotbugsScan.FinalOutput = spotBugsOutput
+
+	// check results and prepare all vulnerabilities found
+	spotbugsScan.prepareSpotBugsVulns()
+	spotbugsScan.prepareContainerAfterScan()
+	return nil
+}
+
+func parseXMLtoJSON(byteValue []byte) (SpotBugsOutput, error) {
+	var bugs SpotBugsOutput
+	if err := xml.Unmarshal(byteValue, &bugs); err != nil {
+		return bugs, err
+	}
+	numOfErrors, err := strconv.Atoi(bugs.Errors.Errors)
+	if err != nil {
+		return bugs, err
+	}
+	numOfMissingClasses, err := strconv.Atoi(bugs.Errors.MissingClasses)
+	if err != nil {
+		return bugs, err
+	}
+	if (len(bugs.SpotBugsIssue) == 0) && (numOfErrors > 0 || numOfMissingClasses > 0) {
+		return bugs, fmt.Errorf("spotbugs has risen errors because of [%d] missing classes and [%d] errors while analysing", numOfMissingClasses, numOfErrors)
+	}
+	return bugs, nil
+}
+
+func (spotbugsScan *SecTestScanInfo) prepareSpotBugsVulns() {
+	var spotbugsOutput SpotBugsOutput
+
+	huskyCIspotbugsResults := types.HuskyCISecurityTestOutput{}
+	spotbugsOutput = spotbugsScan.FinalOutput.(SpotBugsOutput)
+
+	if spotbugsScan.ErrorFound != nil {
+		spotbugsVuln := types.HuskyCIVulnerability{}
+		spotbugsVuln.Language = "Java"
+		spotbugsVuln.SecurityTool = "SpotBugs"
+		spotbugsVuln.Details = fmt.Sprintf("An error occured running huskyCI scan on your Java project: %s", spotbugsScan.ErrorFound.Error())
+		spotbugsVuln.Severity = "LOW"
+		spotbugsVuln.Confidence = "HIGH"
+
+		huskyCIspotbugsResults.LowVulns = append(huskyCIspotbugsResults.LowVulns, spotbugsVuln)
```

Comment:
Hey, @abzcoding! Sorry! I have sent you the wrong code. Could you replace: 

```
huskyCIspotbugsResults.LowVulns = append(huskyCIspotbugsResults.LowVulns, spotbugsVuln)
```

Into this?

```
spotbugsScan.Vulnerabilities.LowVulns = append(spotbugsScan.Vulnerabilities.LowVulns, spotbugsVuln)
```

Reply:
yeah @rafaveira3 my bad 😅